src/uct/sm/mm/base/mm_ep.h

```diff
@@ -43,6 +44,13 @@ typedef struct uct_mm_ep {
         struct sockaddr_un     sockaddr;  /* address of signaling socket */
         socklen_t              addrlen;   /* address length of signaling socket */
     } signal;
+
+    /* owner of segment process information. we have to cache this value
+     * because some transports terminate segment when process gone (xpmem) */
+    struct {
+        char                  proc[UCT_SM_PROC_DIR_NAME_MAX_LEN]; /* Process owner proc dir */
```

Comment:
maybe the proc path could be allocated dynamically, and created only if err handling is needed?
also this should be moved to sm/base

Reply:
there is no info about err handling - err handling is UCP level.
for now not all shm-based transports have keepalive feature that is why starttime is not moved into sm/base object.
do you think it should be moved?